### PR TITLE
Stop logging IPs to storage

### LINKED ISSUE #294: Stop logging IP addresses
Right now, we are logging IP addresses of users in several locations:

- `journalctl -f -u dovecot`
- `journalctl -u postfix@-.service -f`
- `journalctl --facility=2` in general (showing all mail related log messages in journald)
- `/var/log/nginx/access.log`
- `/var/log/nginx/error.log`
- ...other places? don't think so.

postfix (and dovecot as well I think) doesn't allow to specify a log format. Three possible solutions:

1. it allows to specify a log file though. Maybe we could write a log daemon which receives postifx logs and rewrites them before saving them to disk/journald.
2. we could use nginx in stream mode as a reverse TCP proxy, at least on the port(s) Delta Chat connects to. Then all IPs postfix sees would be "localhost". Could this cause problems with validating MTA-STS or reverse DNS lookups though?
3. we could pass the log lines to rsyslog, use an rsyslog config template to scrub the IP addresses from the log line... and maybe even pass them to journald again in the end?

### PR DESCRIPTION
**Pull request by hagenest**
closes #294

### CODE REVIEW

**Comment by hagenest** [comment]
tested deployment on c2.testrun.org manually, nginx logs are now in journald. But: journald is still keeping logs, even after a reboot :| I'll investigate

**Comment by hagenest** [comment]
nvm, it actually works, but it still showed the logs created before setting journalctl to volatile

**Comment by hagenest** [comment]
missed one commit, was part of the test tho

File: cmdeploy/src/cmdeploy/nginx/nginx.conf.j2 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 user www-data;
 worker_processes auto;
 pid /run/nginx.pid;
-error_log /var/log/nginx/error.log;
+error_log stderr;
```
[the remark this answers is not in the document]
Comment (hagenest) [reply cmdeploy/src/cmdeploy/nginx/nginx.conf.j2:4]: stderr goes automatically into journald, but with the correct facility as far as I understood it

[the remark this answers is not in the document]
Comment (hagenest) [reply cmdeploy/src/cmdeploy/nginx/nginx.conf.j2:4]: so errors are facility 3 (error) and access_logs are per default facility local7 (news)

[the remark this answers is not in the document]
Comment (hagenest) [reply cmdeploy/src/cmdeploy/nginx/nginx.conf.j2:4]: also fine by me. I'll change it

[the remark this answers is not in the document]
Comment (hagenest) [reply cmdeploy/src/cmdeploy/nginx/nginx.conf.j2:4]: from the docs:

> The facility= parameter specifies the type of program that is logging the message. The default value is local7. Other possible values are: auth, authpriv, daemon, cron, ftp, lpr, kern, mail, news, syslog, user, uucp, local0 ... local7.

error is not a possible value, it seems, so I think putting it into stderr is the recommended way. It's also the way mentioned in the docs:

https://docs.nginx.com/nginx/admin-guide/monitoring/logging/

[the remark this answers is not in the document]
Comment (hagenest) [reply cmdeploy/src/cmdeploy/nginx/nginx.conf.j2:4]: forget what I wrote

[the remark this answers is not in the document]
Comment (hagenest) [reply cmdeploy/src/cmdeploy/nginx/nginx.conf.j2]: done

**Review by missytake — APPROVED** [review]

**Pull request merged**